### ISSUE TITLE
Consider "JEP 403: Strongly Encapsulate JDK Internals"

### ISSUE THREAD

**Issue by helmbold** [google/google-java-format#662]
To be able to [use the formatter with Java 16](https://github.com/google/google-java-format#jdk-16), one needs to provide some JVM arguments:

```
java \
  --add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
  --add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
  --add-exports jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED \
  --add-exports jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
  --add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
  -jar google-java-format-1.11.0-all-deps.jar <options> [files...]
```

This approach works with [JEP 396: Strongly Encapsulate JDK Internals by Default](https://openjdk.java.net/jeps/396) (JDK 16) but not with [JEP 403: Strongly Encapsulate JDK Internals](https://openjdk.java.net/jeps/403) (JDK 17):

> Strongly encapsulate all internal elements of the JDK, except for critical internal APIs such as sun.misc.Unsafe. It will **no longer be possible** to relax the strong encapsulation of internal elements via a single command-line option, as was possible in JDK 9 through JDK 16.

**Comment by sormuras** [comment]
It works with Java 17 and also with Java 18-ea.

**Comment by tbroyer** [comment]
JEP 403 is only about `--illegal-access`, not about `--add-exports` and/or `--add-opens`.

**Comment by cushon** [comment]
Right, it's not longer possible "via a single command-line option" (`--illegal-access`), it's still possible by passing multiple `-add-exports` / `-add-opens` options. As far as I know there are no plans to remove support for `-add-exports` / `-add-opens`.

It'd be nice to avoid the long command line, but I don't know of any good alternatives.

We should probably think about providing a standard wrapper script for the formatter that sets up the command line.

**Comment by sparty02** [comment]
Since the --add-exports are necessary with Java 16+, how would you set these when using the Eclipse plugin?

**Comment by cushon** [comment]
> how would you set these when using the Eclipse plugin?

Let's track that in https://github.com/google/google-java-format/issues/665, thanks for filing that bug
